news/models.py

```diff
@@ -177,17 +185,54 @@ def copy_relations(self, oldinstance):
         """
         self.tags.set(oldinstance.tags.all())
 
-    def get_latest_articles(self, limit=3):
+    @property
+    def category_articles(self):
+        """
+        Return a queryset for a category
+        """
+        return (
+            Article.objects.published()
+            .filter(category__in=self.category)
+            .distinct()
+            .order_by("-created_at")[: self.num_articles]
+        )
+
+    @property
+    def tags_articles(self):
         """
-        Return a specified number of latest articles based on the tags selected
+        Return a queryset for specified tags
         """
         return (
             Article.objects.published()
             .filter(tags__in=self.tags.all())
             .distinct()
-            .order_by("-created_at")[:limit]
+            .order_by("-created_at")[: self.num_articles]
         )
 
+    @property
+    def articles_recent(self):
+        """
+        Return a default queryset
+        """
+        return (
+            Article.objects.published()
+            .distinct()
+            .order_by("-created_at")[: self.num_articles]
+        )
+
+    def get_articles(self):
+        """
+        Return a queryset based on what the user chooses on the frontend
+        """
+        queryset = self.articles_recent
+
+        if self.tags.all():
+            queryset = self.tags_articles
+        if self.category:
```

Comment:
So this is an either/or e.g if you have a category set, that will override your previous filtering by tags? So you cant have articles filtered by both? Is that the intention?

Reply:
It's the intention for now yes. There's a priority in what queryset is returned, although I should add help text so the user knows

Reply:
Ok yeah - if thats what they want thats fine, but update the helptext to explain how it works. If you did want to filter by both, it wouldnt need too much to update, you would just want to pass your initial `queryset` in to the filter methods